.github/workflows/docs.yml

```diff
@@ -11,15 +11,27 @@ jobs:
       - uses: actions/checkout@v2
         with:
           submodules: 'recursive'
-      - uses: actions/setup-python@v1
+      - uses: actions/setup-python@v2
         with:
-          python-version: "3.7"
-      - name: Install dependencies
+          python-version: "3.10"
+      - name: Bootstrap poetry
         run: |
-          python -m pip install --upgrade pip
-          pip install tox virtualenv
+          curl -sL https://install.python-poetry.org | python - -y ${{ matrix.bootstrap-args }}
+      - name: Configure poetry
+        run: poetry config virtualenvs.in-project true
```

Comment:
Might want to consider using a [dedicated GH action](https://github.com/snok/install-poetry) for that:

```yaml
- name: Install and configure Poetry
  uses: snok/install-poetry@v1
  with:
    virtualenvs-in-project: false
```

Unless there is a good reason not to?

Reply:
I have already discussed that in my last meeting with Audrey and Daniel, we might use this packed action in a separate pull request, but for now, we will go with these multi-steps workflow.
btw, I already tried this action from SNOK and it seems it worked fine.